src/priority/priority.py

```diff
@@ -295,6 +295,11 @@ def __init__(self, maximum_streams=1000):
         self._root_stream = Stream(stream_id=0, weight=1)
         self._root_stream.active = False
         self._streams = {0: self._root_stream}
+
+        if not isinstance(maximum_streams, int):
```

Comment:
Also saw you chose ``int`` here instead of ``six.integer_types`` but I think it's a safe bet no one will want to insert that many streams into a tree.